src/middlewared/middlewared/plugins/system.py

```diff
@@ -23,6 +24,165 @@
+        config_data = await self.config()
+        original_data = config_data.copy()
+        config_data.update(data)
+
+        verrors, config_data = await self.validate_fields('advanced_settings_update', config_data)
+        if verrors:
+            raise verrors
+
+        if len(set(config_data.items()) ^ set(original_data.items())) > 0:
+            if original_data.get('sed_user'):
+                original_data['sed_user'] = original_data['sed_user'].lower()
+            if config_data.get('sed_user'):
+                config_data['sed_user'] = config_data['sed_user'].lower()
+
+            await self.middleware.call(
+                'datastore.update',
+                self._config.datastore,
+                config_data['id'],
+                config_data,
+                {'prefix': self._config.datastore_prefix}
+            )
+
+            loader_reloaded = False
+            if original_data['motd'] != config_data['motd']:
+                await self.middleware.call('service.start', 'motd', {'onetime': False})
+
+            if original_data['consolemenu'] != config_data['consolemenu']:
+                await self.middleware.call('service.start', 'ttys', {'onetime': False})
+
+            if original_data['powerdaemon'] != config_data['powerdaemon']:
+                await self.middleware.call('service.restart', 'powerd', {'onetime': False})
+
+            if original_data['serialconsole'] != config_data['serialconsole']:
+                await self.middleware.call('service.start', 'ttys', {'onetime': False})
+                if not loader_reloaded:
+                    await self.middleware.call('service.reload', 'loader', {'onetime': False})
+                    loader_reloaded = True
+            elif (
+                    original_data['serialspeed'] != config_data['serialspeed'] or
+                    original_data['serialport'] != config_data['serialport']
+            ):
+                if not loader_reloaded:
+                    await self.middleware.call('service.reload', 'loader', {'onetime': False})
+                    loader_reloaded = True
+
+            if original_data['consolescreensaver'] != config_data['consolescreensaver']:
+                if config_data['consolescreensaver'] == 0:
+                    await self.middleware.call('service.stop', 'saver', {'onetime': False})
+                else:
+                    await self.middleware.call('service.start', 'saver', {'onetime': False})
+                if not loader_reloaded:
+                    await self.middleware.call('service.reload', 'loader', {'onetime': False})
+                    loader_reloaded = True
+
+            if (
+                original_data['autotune'] != config_data['autotune'] and
+                not loader_reloaded
+            ):
+                await self.middleware.call('service.reload', 'loader', {'onetime': False})
+
+            if original_data['debugkernel'] != config_data['debugkernel']:
```

Comment:
i think we can improve this bit compared to what old UI was doing. It can reload loader twice like it currently is. How about we add loader_reloaded = True above for autotune, and add `and not loader_reloaded` in this if statement.